wled00/bus_manager.cpp

```diff
@@ -499,46 +539,92 @@ uint32_t BusPwm::getPixelColor(uint16_t pix) {
 
 void BusPwm::show() {
   if (!_valid) return;
-  unsigned numPins = NUM_PWM_PINS(_type);
-  unsigned maxBri = (1<<_depth) - 1;
-  // use CIE brightness formula
-  unsigned pwmBri = (unsigned)_bri * 100;  
-  if(pwmBri < 2040) pwmBri = ((pwmBri << _depth) + 115043) / 230087; //adding '0.5' before division for correct rounding
-  else {  
+  // if _needsRefresh is true (UI hack) we are using dithering (credit @dedehai & @zalatnaicsongor)
+  // https://github.com/Aircoookie/WLED/pull/4115 and https://github.com/zalatnaicsongor/WLED/pull/1)
+  const bool     dithering = _needsRefresh; // avoid working with bitfield
+  const unsigned numPins = getPins();
+  const unsigned maxBri = (1<<_depth);      // possible values: 16384 (14), 8192 (13), 4096 (12), 2048 (11), 1024 (10), 512 (9) and 256 (8) 
+  [[maybe_unused]] const unsigned bitShift = dithering * 4;  // if dithering, _depth is 12 bit but LEDC channel is set to 8 bit (using 4 fractional bits)
+
+  // use CIE brightness formula (cubic) to fit (or approximate linearity of) human eye perceived brightness
+  // the formula is based on 12 bit resolution as there is no need for greater precision
+  // see: https://en.wikipedia.org/wiki/Lightness
+  unsigned pwmBri = (unsigned)_bri * 100;  // enlarge to use integer math for linear response
+  if (pwmBri < 2040) {
+    // linear response for values [0-20]
+    pwmBri = ((pwmBri << 12) + 115043) / 230087; //adding '0.5' before division for correct rounding
+  } else {
+    // cubic response for values [21-255]
     pwmBri += 4080;
-    float temp = (float)pwmBri / 29580;
-    temp = temp * temp * temp * (1<<_depth) - 1; 
-    pwmBri = (unsigned)temp;
-  }
+    float temp = (float)pwmBri / 29580.0f;
+    temp = temp * temp * temp * (float)maxBri; 
+    pwmBri = (unsigned)temp;  // pwmBri is in range [0-maxBri] 
+  }
+
+  [[maybe_unused]] unsigned hPoint = 0;  // phase shift (0 - maxBri)
+  // we will be phase shifting every channel by previous pulse length (plus dead time if required)
+  // phase shifting is only mandatory when using H-bridge to drive reverse-polarity PWM CCT (2 wire) LED type 
+  // CCT additive blending must be 0 (WW & CW will not overlap) otherwise signals *will* overlap
+  // for all other cases it will just try to "spread" the load on PSU
+  // Phase shifting requires that LEDC timers are synchronised (see setup()). For PWM CCT (and H-bridge) it is
+  // also mandatory that both channels use the same timer (pinManager takes care of that).
   for (unsigned i = 0; i < numPins; i++) {
-    unsigned scaled = (_data[i] * pwmBri) / 255;
-    if (_reversed) scaled = maxBri - scaled;
+    unsigned duty = (_data[i] * pwmBri) / 255;    
     #ifdef ESP8266
-    analogWrite(_pins[i], scaled);
+    if (_reversed) duty = maxBri - duty;
+    analogWrite(_pins[i], duty);
```

Comment:
Can we open an issue to track ESP8266 phase support?  I'll take that one on once this merge lands.

Reply:
By all means! 😄